pygmt/helpers/utils.py

```diff
@@ -14,38 +14,43 @@
 from pygmt.exceptions import GMTInvalidInput
 
 
-def data_kind(data, x=None, y=None, z=None, required_z=False):
+def data_kind(data, x=None, y=None, z=None, required_z=False, optional_data=False):
     """
     Check what kind of data is provided to a module.
 
     Possible types:
 
     * a file name provided as 'data'
-    * a pathlib.Path provided as 'data'
-    * an xarray.DataArray provided as 'data'
-    * a matrix provided as 'data'
+    * a pathlib.PurePath object provided as 'data'
+    * an xarray.DataArray object provided as 'data'
+    * a 2-D matrix provided as 'data'
     * 1-D arrays x and y (and z, optionally)
+    * an optional argument (None, bool, int or float) provided as 'data'
 
     Arguments should be ``None`` if not used. If doesn't fit any of these
     categories (or fits more than one), will raise an exception.
 
     Parameters
     ----------
-    data : str or pathlib.Path or xarray.DataArray or {table-like} or None
+    data : str, pathlib.PurePath, None, bool, xarray.DataArray or {table-like}
         Pass in either a file name or :class:`pathlib.Path` to an ASCII data
         table, an :class:`xarray.DataArray`, a 1-D/2-D
-        {table-classes}.
+        {table-classes} or an option argument.
     x/y : 1-D arrays or None
         x and y columns as numpy arrays.
     z : 1-D array or None
         z column as numpy array. To be used optionally when x and y are given.
     required_z : bool
         State whether the 'z' column is required.
+    optional_data : bool
+        State whether 'data' is optional (useful for dealing with optional
+        virtual files).
```

Comment:
Currently the function has `required_z` and `optional_data` parameters. Should we use `required_z` and `required_data` instead? Or `optional_z` and `optional_data`?

Reply:
Actually, I'm thinking about if we should remove `required_z` and use `required_ncols` instead. For example, in `Figure.plot()`, sometimes we need more than three columns (e.g., x, y, size, z_for_color).

Reply:
Right, for #1076/#1117 you mean? Should probably do that in a separate PR.

Reply:
What are the valid values in the list? `color`, `size`, `symbol` and more? I think it will make the codes more complicated.

Reply:
> Currently the function has `required_z` and `optional_data` parameters. Should we use `required_z` and `required_data` instead? Or `optional_z` and `optional_data`?

Renamed `optional_data` to `required_data` in 932d07d5abea788b3ca4aa0a526fe124d01f3020.